### PR TITLE
fix: Add predicate get contact

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

File: MailCore/Cache/ContactManager/ContactManager+DB.swift [hunk 1]
```diff
@@ -70,7 +70,7 @@ public extension ContactManager {
     func getContact(for correspondent: any Correspondent, transactionable: Transactionable) -> MergedContact? {
         transactionable.fetchObject(ofType: MergedContact.self) { partial in
             let matched = partial.where { $0.email == correspondent.email }
-            let result = matched.first { $0.name.caseInsensitiveCompare(correspondent.name) == .orderedSame } ?? matched.first
+            let result = matched.filter(NSPredicate(format: "name ==[c] %@", correspondent.name)).first ?? matched.first
```
Comment (adrien-coye) [inline MailCore/Cache/ContactManager/ContactManager+DB.swift:73]: I understand we cannot chain filtering as we have a fallback to `matched.first`

Still this should work. 
```suggestion
            let result = matched.filter("name ==[c] %@", correspondent.name).first ?? matched.first
```

**Review by adrien-coye — COMMENTED** [review]
Two tiny remarks 🙏

File: MailTests/Contacts/UTContactManager.swift [hunk 2]
```diff
@@ -45,6 +45,24 @@ final class UTContactManager: XCTestCase {
...
+            contact.name = name
+            contact.email = email
+
+            try contactManager.writeTransaction { writableRealm in
+                writableRealm.add(contact)
+            }
+
+            return contact
+        } catch {
+            fatalError("failed transaction in base, error:\(error)")
```
[the remark this answers is not in the document]
Comment (PhilippeWeidmann) [reply MailTests/Contacts/UTContactManager.swift:62]: This is a test 😬

File: MailCore/Cache/ContactManager/ContactManager+DB.swift (hunk 1 above)
Comment (PhilippeWeidmann) [reply MailCore/Cache/ContactManager/ContactManager+DB.swift]: Didn't know about this shortcut. Fixed 👍

File: MailTests/Contacts/UTContactManager.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrien-coye) [reply MailTests/Contacts/UTContactManager.swift:62]: Woops my bad.

**Review by adrien-coye — APPROVED** [review]
Ty 🤝

**Pull request merged**